core/corehttp/gateway_handler_unixfs_file.go

```diff
@@ -37,6 +37,13 @@ func (i *gatewayHandler) serveFile(ctx context.Context, w http.ResponseWriter, r
 		return
 	}
 
+	if size == 0 {
+		// We override null files to 200 to avoid issues with fragment caching reverse proxies.
+		// Also whatever you are asking for, it's cheaper to just give you the complete file (nothing).
+		w.WriteHeader(http.StatusOK)
```

Comment:
We need to set explicit Content-Type to be sure we don't change the current behavior in web browsers (rendering empty page instead of showing "Save As" dialog):

```suggestion
        w.Header().Set("Content-Type",  "text/plain")
		w.WriteHeader(http.StatusOK)
```

Reply:
Thx I didn't knew which type should be there.